### PR TITLE
Upgrade client core versions

### LINKED ISSUE #393: Upgrade angular core

### PR DESCRIPTION
**Pull request by DavidBronfen**
**What kind of change does this PR introduce?** (check _one_ with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Docs related changes
[X] Other... Please describe:
```

**Link to the open issue**:
Fix #393 

**Other information**:
Upgrading all core libraries.

### CODE REVIEW

**Comment by RoySegall** [comment]
@DavidBronfen is that suppose to fix the errors I'm getting in [here](https://github.com/hasadna/open_pension/pull/403#issuecomment-446616527)?

**Comment by RoySegall** [comment]
@DavidBronfen The tests won't pass since the fix for the tests exists in the PR I linked above. If this changes fixes the problems in the docker branch commit them there and I'll try to to see if they fixed the problems or not.

**Comment by RoySegall** [comment]
@DavidBronfen You are in a good direction. I'm not getting the other error but I'm getting othe erros:
```bash
ERROR in ../@ngx-share/button/ngx-share-button.ts(14,5): Error during template compile of 'ShareButtonModule'
  Could not resolve @fortawesome/angular-fontawesome relative to [object Object]..
node_modules/rxjs/Observable.d.ts(1,15): error TS2307: Cannot find module 'rxjs-compat/Observable'.
src/app/components/filters/filters.component.ts(2,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/components/search/search.component.ts(4,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
node_modules/@types/d3-shape/index.d.ts(2273,19): error TS2304: Cannot find name 'CanvasPathMethods'.
src/app/components/pai/pai.component.ts(2,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
node_modules/rxjs/observable/of.d.ts(1,15): error TS2307: Cannot find module 'rxjs-compat/observable/of'.
src/app/services/pai.service.ts(3,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/effects/pai.effect.ts(5,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/observable/of"' has no exported member 'of'.
src/app/effects/pai.effect.ts(7,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/effects/pai.effect.ts(34,6): error TS2339: Property 'ofType' does not exist on type 'Actions<Action>'.
src/app/effects/pai.effect.ts(42,6): error TS2339: Property 'ofType' does not exist on type 'Actions<Action>'.
src/app/effects/pai.effect.ts(50,6): error TS2339: Property 'ofType' does not exist on type 'Actions<Action>'.
src/app/effects/pai.effect.ts(58,6): error TS2339: Property 'ofType' does not exist on type 'Actions<Action>'.
src/app/services/filters.service.ts(2,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/effects/filters.effect.ts(5,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/observable/of"' has no exported member 'of'.
src/app/effects/filters.effect.ts(7,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/effects/filters.effect.ts(27,6): error TS2339: Property 'ofType' does not exist on type 'Actions<Action>'.
src/app/services/quarters.service.ts(2,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/effects/quarters.effect.ts(5,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/observable/of"' has no exported member 'of'.
src/app/effects/quarters.effect.ts(7,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/effects/quarters.effect.ts(31,6): error TS2339: Property 'ofType' does not exist on type 'Actions<Action>'.
src/app/effects/quarters.effect.ts(39,6): error TS2339: Property 'ofType' does not exist on type 'Actions<Action>'.
src/app/services/search.service.ts(2,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/effects/search.effect.ts(2,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/observable/of"' has no exported member 'of'.
src/app/effects/search.effect.ts(4,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/effects/search.effect.ts(31,49): error TS2345: Argument of type '{}' is not assignable to parameter of type 'SearchResult[]'.
  Property 'length' is missing in type '{}'.
src/app/modules/blog/components/blog/blog.component.ts(3,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/modules/blog/components/detail-post/detail-post.component.ts(3,10): error TS2724: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/@ngx-share/core/ngx-share-core"' has no exported member 'ShareButtons'. Did you mean 'IShareButtons'?
src/app/modules/blog/components/detail-post/detail-post.component.ts(4,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/modules/blog/services/post.service.ts(2,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/modules/blog/effects/post.effect.ts(5,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/observable/of"' has no exported member 'of'.
src/app/modules/blog/effects/post.effect.ts(7,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/modules/blog/effects/post.effect.ts(31,6): error TS2339: Property 'ofType' does not exist on type 'Actions<Action>'.
src/app/modules/blog/effects/post.effect.ts(39,6): error TS2339: Property 'ofType' does not exist on type 'Actions<Action>'.
src/app/modules/blog/effects/post.effect.ts(47,6): error TS2339: Property 'ofType' does not exist on type 'Actions<Action>'.
src/app/modules/contact/services/contact.service.ts(2,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/modules/contact/effects/contact.effect.ts(2,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/observable/of"' has no exported member 'of'.
src/app/modules/contact/effects/contact.effect.ts(4,10): error TS2305: Module '"/Applications/MAMP/htdocs/open_pension/client/node_modules/rxjs/Observable"' has no exported member 'Observable'.
src/app/modules/contact/effects/contact.effect.ts(31,69): error TS2345: Argument of type '{}' is not assignable to parameter of type 'Contact'.
  Property 'name' is missing in type '{}'.

ERROR in ./src/styles.scss (./node_modules/@angular-devkit/build-angular/src/angular-cli-files/plugins/raw-css-loader.js!./node_modules/postcss-loader/src??embedded!./node_modules/sass-loader/lib/loader.js??ref--14-3!./src/styles.scss)
Module build failed (from ./node_modules/sass-loader/lib/loader.js):

@import "~@ngx-share/button/styles/share-buttons";
^
      File to import not found or unreadable: ~@ngx-share/button/styles/share-buttons.
      in /Applications/MAMP/htdocs/open_pension/client/src/styles.scss (line 4, column 1)
ℹ ｢wdm｣: Failed to compile.

```

And after a fresh install I'm getting:
```bash
node-pre-gyp ERR! Tried to download(404): https://fsevents-binaries.s3-us-west-2.amazonaws.com/v1.1.3/fse-v1.1.3-node-v64-darwin-x64.tar.gz 
node-pre-gyp ERR! Pre-built binaries not found for fsevents@1.1.3 and node@10.11.0 (node-v64 ABI, unknown) (falling back to source compile with node-gyp) 
node-pre-gyp ERR! Tried to download(undefined): https://fsevents-binaries.s3-us-west-2.amazonaws.com/v1.1.3/fse-v1.1.3-node-v64-darwin-x64.tar.gz 
node-pre-gyp ERR! Pre-built binaries not found for fsevents@1.1.3 and node@10.11.0 (node-v64 ABI, unknown) (falling back to source compile with node-gyp) 
````

But I think that's minor compare to the first errors I pasted.

**Comment by DavidBronfen** [comment]
@RoySegall 
So as I told you in PM this PR is about to upgrade the client libraries.
The errors related to some wrong imports that occurred after I made some upgrades of some client libraries.
I will take care of it :)

**Comment by RoySegall** [comment]
@DavidBronfen Look at you, success:
image

**Comment by RoySegall** [comment]
@DavidBronfen I'm still getting more errors while installing:
```
gyp  SOLINK_MODULE(target) Release/fse.node
 ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Darwin 18.2.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/Applications/MAMP/htdocs/open_pension/client/node_modules/fsevents/lib/binding/Release/node-v64-darwin-x64/fse.node" "--module_name=fse" "--module_path=/Applications/MAMP/htdocs/open_pension/client/node_modules/fsevents/lib/binding/Release/node-v64-darwin-x64"
gyp ERR! cwd /Applications/MAMP/htdocs/open_pension/client/node_modules/fsevents
gyp ERR! node -v v10.11.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Applications/MAMP/htdocs/open_pension/client/node_modules/fsevents/lib/binding/Release/node-v64-darwin-x64/fse.node --module_name=fse --module_path=/Applications/MAMP/htdocs/open_pension/client/node_modules/fsevents/lib/binding/Release/node-v64-darwin-x64' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Applications/MAMP/htdocs/open_pension/client/node_modules/fsevents/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:962:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
node-pre-gyp ERR! System Darwin 18.2.0
node-pre-gyp ERR! command "/usr/local/bin/node" "/Applications/MAMP/htdocs/open_pension/client/node_modules/fsevents/node_modules/node-pre-gyp/bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /Applications/MAMP/htdocs/open_pension/client/node_modules/fsevents
node-pre-gyp ERR! node -v v10.11.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.39
node-pre-gyp ERR! not ok 

```

and 
```
added 1789 packages from 1238 contributors and audited 26103 packages in 26.875s
found 18 vulnerabilities (3 low, 9 moderate, 6 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

**Comment by RoySegall** [comment]
@DavidBronfen And of course - don't forget to merge master. Iv'e merged #402

File: client/src/app/components/pai/pai.component.ts [hunk 1]
```diff
@@ -84,7 +84,8 @@ export class PaiComponent implements OnInit, OnDestroy {
 
   loadData(root) {
     const partition = d3partition();
-    const color = scale.scaleOrdinal(scale.schemeCategory20);
+    // TODO Change the schemeCategory20 to the new
```
Comment (ItamarGronich) [inline client/src/app/components/pai/pai.component.ts]: @DavidBronfen We talked about this.
Let's not forget to do this before we merge.

File: client/src/app/components/pai/pai.component.ts [hunk 2]
```diff
@@ -102,7 +103,9 @@ export class PaiComponent implements OnInit, OnDestroy {
       .style('fill', (d: any) => {
         const colorNode = this.selectedFilters.filter((node, index) => (d.depth) === index);
         if (colorNode.length) {
-          return colorNode[0].color;
+          // Old return based on schemeCategory20 colors that is deprecated in this d3 version
```
Comment (ItamarGronich) [inline client/src/app/components/pai/pai.component.ts]: Same here

File: client/src/polyfills.ts [hunk 3]
```diff
@@ -43,7 +43,7 @@
 
 /** Evergreen browsers require these. **/
 // Used for reflect-metadata in JIT. If you use AOT (and only Angular decorators), you can remove.
```
Comment (ItamarGronich) [inline client/src/polyfills.ts]: Let's remove this comment also.

File: client/tsconfig.json [hunk 4]
```diff
@@ -14,6 +15,8 @@
     "lib": [
       "es2017",
       "dom"
-    ]
+    ],
+    "module": "es2015",
+    "baseUrl": "./"
   }
-}
+}
```
Comment (ItamarGronich) [inline client/tsconfig.json]: new line missing here.

**Pull request merged**